src/rdkafka_partition.c

```diff
@@ -609,26 +609,23 @@ void rd_kafka_toppar_desired_del (rd_kafka_toppar_t *rktp) {
  * Append message at tail of 'rktp' message queue.
  */
 void rd_kafka_toppar_enq_msg (rd_kafka_toppar_t *rktp, rd_kafka_msg_t *rkm) {
+	int wakeup_fd, queue_len;
+	rd_kafka_t *rk;
 
 	rd_kafka_toppar_lock(rktp);
+	wakeup_fd = rktp->rktp_msgq_wakeup_fd;
+	rk = rktp->rktp_rkt->rkt_rk;
 	rd_kafka_msgq_enq(&rktp->rktp_msgq, rkm);
+	queue_len = rd_kafka_msgq_len(&rktp->rktp_msgq);
+        rd_kafka_toppar_unlock(rktp);
 #ifndef _MSC_VER
-        if (rktp->rktp_msgq_wakeup_fd != -1 &&
-            rd_kafka_msgq_len(&rktp->rktp_msgq) == 1) {
+        if (wakeup_fd >= 0 && queue_len == 1) {
```

Comment:
Please try to keep as much of the existing code as possible if a change is not strictly needed.
E.g.. `wakeup_fd == -1 ..`

I think we should change msgq_len() to return the return-value from atomic_add() so that we don't need the msgq_len() call at all.

Reply:
the change to the comparison was an accident, and I'll revert that, though I do sort of think >= 0 is a better test given that other negative return values don't mean anything.
I was planning on doing that change to msgq_enq, but wanted to keep it separate, and didn't think of it till this was written.